lmdeploy/vl/model/llava_hf.py

```diff
@@ -152,3 +153,33 @@ def to_turbomind(self, messages, chat_template, tokenizer, sequence_start):
                                                  sequence_start)
         return self.to_turbomind_aux(messages, prompt, IMAGE_TOKEN, tokenizer,
                                      sequence_start)
+
+    @staticmethod
+    def get_vision_encoder_device_map(
+            model,
+            max_memory,
+            no_split_module_classes=['CLIPEncoderLayer', 'SiglipEncoderLayer'],
+            same_device_keys=None):
+        """map vision_encoder to same device."""
+        from accelerate.utils import get_balanced_memory, infer_auto_device_map
+        max_memory = get_balanced_memory(
+            model,
+            max_memory=max_memory,
+            dtype=torch.half,
+            no_split_module_classes=no_split_module_classes)
+        device_map = infer_auto_device_map(
+            model,
+            no_split_module_classes=no_split_module_classes,
+            max_memory=max_memory,
+            dtype=torch.half)
+
+        if not same_device_keys:
+            return device_map
+
+        for keys in same_device_keys:
+            fuzzy_keys = [kk for kk in device_map for k in keys if kk.find(k)]
```

Comment:
`fuzzy_keys = [kk for kk in device_map for k in keys if kk.find(k) != -1]` ?

Reply:
`fuzzy_keys = [k for k in keys if k in device_map]`

Reply:
after changing it to 
```
fuzzy_keys = [kk for kk in device_map for k in keys if k in kk]
```
it suffers `RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:1 and cuda:2!`

I will fix it later on